homeassistant/components/firmata/binary_sensor.py

```diff
@@ -0,0 +1,84 @@
+"""Support for Firmata binary sensor input."""
+
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+
+from .board import FirmataBoardPin
+from .const import CONF_NEGATE_STATE, DOMAIN, PIN_MODE_INPUT, PIN_MODE_PULLUP
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Firmata binary sensors."""
+    new_entities = []
+
+    board_name = config_entry.data[CONF_NAME]
+    boards = hass.data[DOMAIN]
+    board = boards[board_name]
+    for binary_sensor in board.binary_sensors:
+        binary_sensor_entity = FirmataDigitalBinaryInput(
+            hass, board_name, **binary_sensor
+        )
+        new_binary_sensor = await binary_sensor_entity.setup_pin()
+        if new_binary_sensor:
+            new_entities.append(binary_sensor_entity)
+
+    async_add_entities(new_entities)
+
+
+class FirmataDigitalBinaryInput(FirmataBoardPin, BinarySensorEntity):
+    """Representation of a Firmata Digital Input Pin."""
+
+    async def setup_pin(self):
+        """Set up a digital input pin."""
+        _LOGGER.debug(
+            "Setting up binary sensor pin %s for board %s", self._name, self._board_name
+        )
+        if not self._mark_pin_used():
+            _LOGGER.warning(
+                "Pin %s already used! Cannot use for binary \
+sensor %s",
+                str(self._pin),
```

Comment:
Logger will already convert to string.
```suggestion
                self._pin,
```